### PR TITLE
diagnosticLog

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: react-native/host/src/events/events.ts [hunk 1]
```diff
@@ -112,6 +112,6 @@ export function addEventListener(listener: (event: MWPEvent) => void): EmitterSu
   return diagnosticLogger.addListener('mwp_event', listener);
 }
 
-export function emitEvent(event: MWPEvent) {
+export function diagnosticLog(event: MWPEvent) {
```
Comment (bangtoven) [inline react-native/host/src/events/events.ts:115]: https://github.com/coinbase/coinbase-wallet-sdk/pull/521
maybe i am too anxious or overcautious about this, but iirc, we had similar discussion on how we'd call the logger object in the web sdk side and avoid any confusion from users. 
curious how you think @vishnumad

Comment (vishnumad) [reply react-native/host/src/events/events.ts:115]: Gotcha, sounds good. Maybe the function `addEventListener` should be renamed to `addDiagnosticListener` or something similar.

**Review by vishnumad — DISMISSED** [review]

File: react-native/host/src/events/events.ts (hunk 1 above)
Comment (bangtoven) [reply react-native/host/src/events/events.ts:115]: great. made the suggested changes as well

File: react-native/client/example/App.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ const App = function () {
 
   useEffect(function setupDeeplinkHandling() {
     // Pass incoming deeplinks into Mobile SDK
-    const subscription = Linking.addEventListener('url', ({url}) => {
+    const subscription = Linking.addDiagnosticLogListener('url', ({url}) => {
```
Comment (vishnumad) [inline react-native/client/example/App.js]: This one shouldn't be changed since it's for incoming deeplinks

**Review by vishnumad — CHANGES_REQUESTED** [review]

**Comment by cb-heimdall** [comment]
Review Error for vishnumad @ 2022-11-22 00:01:35 UTC
User failed mfa authentication, public email is not set on your github profile. see [go/mfa-help](http://go/mfa-help)

File: react-native/client/example/App.js (hunk 2 above)
Comment (bangtoven) [reply react-native/client/example/App.js]: duh... fixed it

**Review by vishnumad — APPROVED** [review]

**Pull request merged**